src/radical/pilot/agent/agent_0.py

```diff
@@ -627,19 +627,21 @@ def _check_control(self, _, msg):
             # we don't handle that request
             return True
 
-        rpc_res = {'uid': req['uid']}
+        rpc_res = {'uid': arg['uid']}
         try:
+            print(arg)
             if req == 'hello'   :
                 ret = 'hello %s' % ' '.join(arg['arg'])
 
             elif req == 'prep_env':
                 env_id   = arg['arg']['env_id']
                 env_spec = arg['arg']['env_spec']
-                ret = self._prepare_env(env_id, env_spec)
+                self._prepare_env(env_id, env_spec)
+                ret = (env_id, env_spec)
 
         except Exception as e:
             # request failed for some reason - indicate error
-            rpc_res['err'] = str(e)
+            rpc_res['err'] = repr(e)
```

Comment:
actually, if revert this back and in test revert back to `'err': "'arg'"`, then it works

Reply:
Which means that `repr` adds a comma on the GH test VM, but not otherwise.  Urgh...  Let me look into that!

Reply:
https://bugs.python.org/issue30399 - this seems to / should have been fixed in 2017...  Hmm...  I don't really want to revert, returning `repr` (and thus the exception type) is actually the right thing to do...

Reply:
it also says about py3.7 only... (even if it would be fixed for 3.7 we still support 3.6 as well)